File: week 3/different_summands/different_summands.cpp
#include <iostream>
#include <vector>

using std::vector;

vector<int> optimal_summands(int n) {
  vector<int> summands;
  if(n<=2) {
	  summands.push_back(n);
  }
  else {
	  int l = 1;
	  while(n > 2*l) {
		  summands.push_back(l);
		  n -= l;
		  l++;
	  }
	  summands.push_back(n);
  }
  return summands;
}

int main() {
  int n;
  std::cin >> n;
  vector<int> summands = optimal_summands(n);
  std::cout << summands.size() << '\n';
  for (size_t i = 0; i < summands.size(); ++i) {
    std::cout << summands[i] << ' ';
  }
}

File: week 3/different_summands/different_summands.py
def optimal_summands(n):
    summands = []
    if(n<=2):
        summands.append(n)
    else:
        l = 1
        while(n > 2*l):
            summands.append(l)
            n -= l
            l += 1
        summands.append(n)
    #write your code here
    return summands

if __name__ == '__main__':
    n = int(raw_input())
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print str(x) + " ",
